### ISSUE TITLE
Flagging returned rows

### ISSUE THREAD

**Issue by bentcoder** [jingyugao/rowserrcheck#32]
Hi,

Both `Query` calls in the function gets flagged. I cannot close the rows in this function because I am reading and closing it in actual caller function instead. Is this expected behaviour?

Thanks

```go
import (
	"context"
	"errors"

	"github.com/jackc/pgx/v5"
	"github.com/jackc/pgx/v5/pgconn"
	"github.com/jackc/pgx/v5/pgxpool"
)

type Query struct {
	TXN pgx.Tx
	DTB    *pgxpool.Pool
}

func (q Query) Run(ctx context.Context, sql string, args ...any) (pgx.Rows, error) {
	switch {
	case q.TXN != nil:
		return q.TXN.Query(ctx, sql, args...)
	case q.DTB != nil:
		return q.DTB.Query(ctx, sql, args...)
	}

	return nil, errors.New("missing runner")
}
```

**Comment by miparnisari** [comment]
+1. This is a false positive that should be ignored because the caller is closing the rows.
